src/features/threadPruner/getThreadStats.ts

```diff
@@ -0,0 +1,43 @@
+import { APIThreadChannel, SnowflakeUtil } from 'discord.js'
+
+export interface PruningCriteria {
+  name: string
+  threadIds: string[]
+}
+
+export function getThreadStats(threads: APIThreadChannel[]) {
+  threads = Array.from(threads).sort(
+    (a, b) => +(b.last_message_id ?? 0) - +(a.last_message_id ?? 0),
+  )
+
+  // Create pruning criteria
+  let pruningCriteria: PruningCriteria[] = []
+
+  // By last message timestamp
+  for (const h of [72, 48, 24, 6, 3, 1]) {
+    const name = `Prune threads with last message older than ${h}h`
+    const threadIds = threads
+      .filter((thread) => {
+        const timestamp = thread.last_message_id
+          ? SnowflakeUtil.timestampFrom(thread.last_message_id)
+          : Infinity
+        return Date.now() - timestamp > 3600e3 * h
+      })
+      .map((t) => t.id)
+    pruningCriteria.push({ name, threadIds })
+  }
+
+  // By number of messages
+  for (const n of [100]) {
+    const name = `Prune bottom ${n} threads`
+    const threadIds = threads
+      .filter((_, index) => index >= threads.length - n)
+      .map((t) => t.id)
+    pruningCriteria.push({ name, threadIds })
+  }
+
+  // Remove empty criteria
+  pruningCriteria = pruningCriteria.filter((c) => c.threadIds.length > 0)
+
+  return { pruningCriteria }
```

Comment:
for functions that calculates statistics, i like returning an object even if it has only one property because:

1. it leaves room for more things or metadata to be added in the future without much refactoring needed. i.e. https://overreacted.io/optimized-for-change/

2. it kinda encourages other places that use this code to use the same name.

   ```js
   // without returning object, the least effort is to name something shorter
   const criteria = getThreadStats(…)

   // with returning object, the least effort is just use the same name
   // it is also clearer that it returns a "stats" object and we are just
   // picking out `pruningCriteria` out of it
   const { pruningCriteria } = getThreadStats(…)

   // if they want shorter name (not encouraged) they have to put some extra effort
   const { pruningCriteria: criteria } = getThreadStats(…)
   ```

3. it allows function name to be more generalized. e.g. `getThreadStats` vs `getPruningCriteria`. using generalized names, when new kind of stats needs to be added, you can just add code to this function, allowing some logic to be reused easily. otherwise, each stat calculating function may have to do redundant work, e.g. sorting threads.